### PR TITLE
(CAS2-208) Use timeline events

### PR DESCRIPTION
**Pull request by libuk**
[Jira ticket](https://dsdmoj.atlassian.net/browse/CAS2-208?atlOrigin=eyJpIjoiODRlNzIwOTlhOTRmNDI2ZGFhNzg3ZmYyMGY3MGQxYjgiLCJwIjoiaiJ9)

# Context
We are expanding the timeline component to show different types of
timeline events: Notes, status updates and when an application has been
created.

# Changes in this PR
* Use `application.timelineEvents` in timeline component
* Feature flag `notes` event type

## Screenshots of UI changes
Screenshot 2024-02-14 at 16 47 35

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/views/partials/_timeline.njk [hunk 1]
```diff
@@ -19,6 +19,14 @@
         <p class="moj-timeline__date">
           <time datetime="{{event.datetime.timestamp}}">{{ event.datetime.date }}</time>
         </p>
+
+        {% if showDescription %}
```
Comment (Harriethw) [inline server/views/partials/_timeline.njk:23]: I'm getting a bit confused following the flag around - I wonder if the logic could be outsourced to the `util` function so there's one place that decides what's in the timeline? And then the view just conditionally renders `if event.description.text` and doesn't have to know about the flag? I guess the description is always going to be nullable so we have to account for that anyway?

**Review by patrickjfl — APPROVED** [review]
LGTM, nice work 👍

File: server/controllers/assess/submittedApplicationsController.ts [hunk 2]
```diff
@@ -41,10 +42,13 @@ export default class SubmittedApplicationsController {
 
       const status = application.statusUpdates.length ? application.statusUpdates[0].label : 'Received'
 
+      const showTimelineDescription = config.flags.notesDisabled === 'false'
```
Comment (Harriethw) [inline server/controllers/assess/submittedApplicationsController.ts:45]: as the `Notes` have already been filtered out in the `utils` is this just hiding the `description` for the status update details, as below? But they aren't behind a feature flag atm right, was there a UCD requirement for them to be hidden?
Screenshot 2024-02-22 at 14 58 01

File: integration_tests/tests/apply/submitted_applications/view_application_overview_as_referrer.cy.ts [hunk 3]
```diff
@@ -41,13 +46,19 @@ context('View submitted application overview', () => {
...
+              label: 'Awaiting decision',
+              body: 'All information has been received and the application is awaiting assessment.',
+            }),
+            timelineEventsFactory.build({
+              occurredAt: DateFormats.dateObjToIsoDateTime(faker.date.past()),
+              createdByName: 'Anne Assessor',
               label: 'More information requested',
-              description:
-                'More information about the application has been requested from the POM (Prison Offender Manager).',
+              body: 'More information about the application has been requested from the POM (Prison Offender Manager).',
```
Comment (Harriethw) [inline integration_tests/tests/apply/submitted_applications/view_application_overview_as_referrer.cy.ts:61]: would it be worth making a few types of timelineEvents, some with `body` and some without?

File: integration_tests/tests/apply/submitted_applications/view_application_overview_as_referrer.cy.ts [hunk 4]
```diff
@@ -41,13 +46,19 @@ context('View submitted application overview', () => {
...
-              updatedAt: DateFormats.dateObjToIsoDateTime(faker.date.past()),
-              updatedBy: externalUserFactory.build({ name: 'Anne Assessor' }),
-              name: 'moreInfoRequested',
+          ],
+          timelineEvents: [
+            timelineEventsFactory.build({
+              occurredAt: DateFormats.dateObjToIsoDateTime(faker.date.future()),
+              createdByName: 'Anne Other Assessor',
+              label: 'Awaiting decision',
+              body: 'All information has been received and the application is awaiting assessment.',
```
Comment (Harriethw) [inline integration_tests/tests/apply/submitted_applications/view_application_overview_as_referrer.cy.ts:55]: Minor but a status update will only have a `body` of status details at the moment, so could update test data to reflect that.

Comment (Harriethw) [reply integration_tests/tests/apply/submitted_applications/view_application_overview_as_referrer.cy.ts:55]: aware that's not really codified anywhere 😢

File: server/views/partials/_timeline.njk (hunk 1 above)
Comment (libuk) [reply server/views/partials/_timeline.njk:23]: That's a lot neater 👍🏾

File: integration_tests/tests/apply/submitted_applications/view_application_overview_as_referrer.cy.ts (hunk 4 above)
Comment (libuk) [reply integration_tests/tests/apply/submitted_applications/view_application_overview_as_referrer.cy.ts:55]: > Minor but a status update will only have a body of status details at the moment, so could update test data to reflect that.

@Harriethw I don't quite follow, could you give me an example?

**Pull request merged**